cli/module_registry.go

```diff
@@ -679,3 +684,78 @@ func createTarballForUpload(moduleUploadPath string, stdout io.Writer) (string,
 	}
 	return tmpFile.Name(), nil
 }
+
+func readModels(path string, logger golog.Logger) ([]ModuleComponent, error) {
```

Comment:
FWIW I wrote [a similar PR](https://github.com/viamrobotics/rdk/pull/3055) to this one a while ago. My `verifyModule` method is similar to your `readModels` method but prints more information to stdout and tries to remove the module as well to ensure it stops correctly. If you'd like to do more granular module verification through mocking a module manager, you're welcome to copy from there. Also feel free to not do that haha, just thought I'd point out.